### ISSUE TITLE
Test parallelisation

### ISSUE THREAD

**Issue by kovalikp** [nuproj/nuproj#99]
I've started to work on enabling parallel test here https://github.com/kovalikp/nuproj/tree/test_pararellism. Tests use random `OutDir` and `IntermediateOutputPath` to isolate individual tests. But I've run into couple of problems:
- I had to set `DisableInProcNode = true` for scenarios builds (csproj, nuproj), otherwise the tests kept throwing errors.
- I cannot get .sln builds running. Out-of-proc setting cause random errors. In-proc under v12 build tools version hang. In-proc under v14 straight out throw an exception. (See methods marked as `Obsolete` here: https://github.com/kovalikp/nuproj/blob/test_pararellism/src/NuProj.Tests/Infrastructure/ScenarioExtensions.cs#L57) @AArnott @terrajobst do you know how to get around this?

Since I've changed tests to use class name for scenario path, and latest xunit runner uses full method name as test name, we might reconsider Scenarios naming convention. I'm thinking about `Solution/ProjectName/ProjectName.ext` = `Class.Method`, so we can keep using `CallerMemberName`.

**Comment by AArnott** [comment]
Have you seen #97 (both the change to get back test method names as we saw them before, and the comments that eventually I think we should switch back).

As for the issues with multi-proc build, yes I know that area of code very well. Executing builds in parallel, but from different tests, will require that each test have its own BuildManager, and yse that you disable the in-proc node so they don't stomp on each other.

**Comment by kovalikp** [comment]
Yep, I've seen #97. I wonder what's your opinion on `IScenario`, `ScenarioBase`, `ScenarioExtensions` implementation, and suggested naming convention. Is it a way to go?

Out of curiosity: Building .sln out-of-proc, is there a way to this?
